src/main/java/org/folio/services/validator/engine/ValidationEngineServiceImpl.java

```diff
@@ -162,44 +164,38 @@ private void validatePasswordByRexExpRule(final String password,
     Promise<JsonObject> promise = Promise.promise();
     String okapiUrl = headers.get(OKAPI_URL_HEADER);
     String userNameRequestUrl = String.format("%s/users?query=id==%s", okapiUrl, userId);
-    HttpClientRequest request = httpClient.getAbs(userNameRequestUrl);
+    HttpRequest<Buffer> request = webClient.getAbs(userNameRequestUrl);
     request
       .putHeader(OKAPI_HEADER_TOKEN, headers.get(OKAPI_HEADER_TOKEN))
       .putHeader(OKAPI_HEADER_TENANT, headers.get(OKAPI_HEADER_TENANT))
       .putHeader(HttpHeaders.CONTENT_TYPE.toString(), MediaType.APPLICATION_JSON)
       .putHeader(HttpHeaders.ACCEPT.toString(), MediaType.APPLICATION_JSON)
-      .handler(response -> {
-        if (response.statusCode() != 200) {
-          response.bodyHandler(buf -> {
-            String message = "Error looking up user at url '" + userNameRequestUrl
-              + "' Expected status code 200, got '" + response.statusCode() +
-              "' :" + buf.toString();
-            promise.fail(message);
-          });
-        } else {
-          response.bodyHandler(buf -> {
-            JsonObject resultObject = buf.toJsonObject();
-            if (!resultObject.containsKey("totalRecords") || !resultObject.containsKey("users")) {
-              promise.fail("Error, missing field(s) 'totalRecords' and/or 'users' in user response object");
+      .send(ar -> {
+        if (ar.succeeded()) {
+          HttpResponse<Buffer> response = ar.result();
+          if (response.statusCode() != 200) {
+            return;
```

Comment:
Should it fail the promis before returning?

Reply:
added, thank you.